### DISCUSSION TITLE
[@types/lodash] _.bind() has only any types but _.partial() has full types

### DISCUSSION THREAD

**Discussion by Hades32** [DefinitelyTyped/DefinitelyTyped#53854] [Issues with a @types package]
I'd like to understand why `_.bind()` doesn't return the same function signature as was the input. Shouldn't the function look the same from the outside?

**Comment by atomanyih** [comment]
@Hades32 Ran into this problem today and cobbled together a type definition based on Variadic Tuples from typescript docs: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html

```ts
declare module 'lodash/bind' {
  type Arr = readonly unknown[];

  export default function bind<T, Head extends Arr, Rest extends Arr, R>(
    fn: (this: T, ...args: [...Head, ...Rest]) => R,
    thisArg: T,
    ...headArgs: Head
  ): (...args: Rest) => R;
}
```

Not a typescript expert, but this seems to work for my project
